src/key.rs

```diff
@@ -418,6 +443,20 @@ impl PublicKey {
         }
     }
 
+    /// Creates a [`PublicKey`] using the key material from `pk` combined with the `parity`.
+    pub fn from_xonly(pk: XOnlyPublicKey, parity: Parity) -> Result<PublicKey, Error> {
+        let mut buf = [0u8; 33];
+
+        // First byte of a compressed key should be `0x02 AND parity`.
+        buf[0] = match parity {
+            Parity::Even => 0x02,
+            Parity::Odd => 0x03,
+        };
+        buf[1..].clone_from_slice(&pk.serialize());
+
+        PublicKey::from_slice(&buf)
```

Comment:
You can safely `unwrap/expect` here.

Reply:
Cool, thanks. Error return removed as suggested.